### PR TITLE
Fix and update api docs

### PR DESCRIPTION
**Pull request by sdepold**

### CODE REVIEW

File: docs/docs-generator.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ if (program.file) {
     {file:'lib/sequelize.js', output: 'sequelize'},
     {file:'lib/instance.js', output: 'instance'},
     {file:'lib/model.js', output: 'model'},
-    {file:'lib/querying.js', output: 'querying'},
```
Comment (sdepold) [inline docs/docs-generator.js]: Not sure what this was meant to be but such a file does not exist

Comment (mickhansen) [inline docs/docs-generator.js]: That's supposed to be `docs/querying`, guess i made an error.

File: docs/api/datatypes.md [hunk 2]
```diff
@@ -97,103 +118,103 @@ Available properties: `UNSIGNED`, `ZEROFILL`
 
 ***
 
-<a name="boolean"></a>
-## `BOOLEAN()`
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/data-types.js#L248)
+<a name="oolea"></a>
+## `OOLEA()`
```
Comment (mickhansen) [inline docs/api/datatypes.md]: OOLEA? ;p

File: docs/api/datatypes.md [hunk 3]
```diff
@@ -77,19 +78,39 @@ Available properties: `UNSIGNED`, `ZEROFILL`
...
+[View code](https://github.com/sequelize/sequelize/blob/a3155ec82d64a76ada2aaaadcb09d4161912819d/lib/data-types.js#L244)
+Floating point number (8-byte precision). Accepts one or two arguments for precision
+
+Available properties: `UNSIGNED`, `ZEROFILL`
+
+
+***
+
+<a name="ecima"></a>
+## `ECIMA()`
```
Comment (mickhansen) [inline docs/api/datatypes.md]: ECIMA?

File: docs/api/datatypes.md [hunk 4]
```diff
@@ -77,19 +78,39 @@ Available properties: `UNSIGNED`, `ZEROFILL`
...
+[View code](https://github.com/sequelize/sequelize/blob/a3155ec82d64a76ada2aaaadcb09d4161912819d/lib/data-types.js#L225)
+Floating point number (4-byte precision). Accepts one or two arguments for precision
+
+Available properties: `UNSIGNED`, `ZEROFILL`
+
+
+***
+
+<a name="oubl"></a>
+## `OUBL()`
```
Comment (mickhansen) [inline docs/api/datatypes.md]: OUBL?

File: docs/api/datatypes.md [hunk 5]
```diff
@@ -50,16 +51,16 @@ Available properties: `BINARY`
 
 ***
 
-<a name="text"></a>
-## `TEXT()`
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/data-types.js#L108)
+<a name="ex"></a>
+## `EX()`
```
Comment (mickhansen) [inline docs/api/datatypes.md]: EX?
THere's a whole bunch of these.

File: docs/api/model.md [hunk 6]
```diff
@@ -9,227 +10,9 @@ A Model represents a table in the database. Sometimes you might also see it refe
 
 ***
 
-<a name="removeattribute"></a>
```
Comment (mickhansen) [inline docs/api/model.md]: Where did this entire block go?

File: docs/api/query-chainer.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
+<a name="querychainer"></a>
```
Comment (mickhansen) [inline docs/api/query-chainer.md]: query chainer is being removed completely in 3.0 so no point in providing docs for it.

File: docs/api/sequelize.md [hunk 8]
```diff
@@ -197,9 +198,9 @@ Emitted when a validation fails
 
 ***
 
-<a name="validationerroritem"></a>
-## `ValidationErrorItem`
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/sequelize.js#L305)
+<a name="alidationerrorite"></a>
+## `alidationErrorIte`
```
Comment (mickhansen) [inline docs/api/sequelize.md]: More weird changes

**Comment by janmeier** [comment]
Yep, definitely some weird stuff going on with dox, I just haven't had time to figure out what...

File: docs/docs-generator.js (hunk 1 above)
Comment (sdepold) [inline docs/docs-generator.js]: Yeah but that file is not generated by the docs-generator but manually authored right?

File: docs/api/query-chainer.md (hunk 7 above)
Comment (sdepold) [inline docs/api/query-chainer.md]: oh yeah this is not supposed to be here

Comment (sdepold) [inline docs/api/query-chainer.md]: fixed

**Comment by sdepold** [comment]
Let's see if I can find out what is going on here

**Comment by sdepold** [comment]
Looks as if it is fixed :)

File: docs/api/model.md (hunk 6 above)
Comment (sdepold) [inline docs/api/model.md]: Good question. No clue (yet)

**Comment by sdepold** [comment]
OK the current code looks good. Let me check what happened to the one block

File: docs/api/model.md (hunk 6 above)
Comment (sdepold) [inline docs/api/model.md]: It seems that the parser is somehow not properly working for the models file

Comment (janmeier) [inline docs/api/model.md]: Maybe try adding a @method removeAttributes or @function removeAttributes tag

**Comment by sdepold** [comment]
So this is part of the models code makes the parser ignore the first ~800 lines of the model https://github.com/sequelize/sequelize/blob/master/lib/model.js#L85-L110

**Comment by sdepold** [comment]
fixed

**Comment by janmeier** [comment]
Great .... 

Perhaps if you can create a smaller test case, post it at removeAttributes

File: docs/api/associations.md [hunk 9]
```diff
@@ -153,7 +153,7 @@ All methods return a promise
 
 <a name="hasmany"></a>
 ## `hasMany(target, [options])`
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/associations/mixin.js#L245)
+[View code](https://github.com/sequelize/sequelize/blob/0c5c8bd7a208d8231ff9d138f369f287ae168cbe/lib/associations/mixin.js#L245)
```
Comment (janmeier) [inline docs/api/associations.md]: I really need to figure out a better solution to this so we don't have to commit sha hash for each method ...

I'm thinking we can add a preprocess step on RTD which inserts this instead

**Comment by sdepold** [comment]
@janmeier @mickhansen ready for review / merge

File: docs/api/associations.md (hunk 9 above)
Comment (sdepold) [inline docs/api/associations.md]: While it would be great I also don't care too much about it

Comment (janmeier) [inline docs/api/associations.md]: But it does make the changes harder to review - a lot of noise :)

**Comment by sdepold** [comment]
Hm. Cannot reproduce the issue in a test case :-|

**Comment by sdepold** [comment]
Ah ! got it!

**Comment by sdepold** [comment]
There we go https://github.com/tj/dox/pull/168

File: docs/api/instance.md [hunk 10]
```diff
@@ -26,14 +26,14 @@ Accessing properties directly or using `get` is preferred for regular use, `getD
...
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/instance.js#L54)
+[View code](https://github.com/sequelize/sequelize/blob/0c5c8bd7a208d8231ff9d138f369f287ae168cbe/lib/instance.js#L54)
 Returns true if this instance has not yet been persisted to the database
 
 ***
 
 <a name="model"></a>
-## `Model()` -> `Model`
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/instance.js#L63)
+## `Model()` -> `ode`
```
Comment (BridgeAR) [inline docs/api/instance.md]: ode

File: docs/api/transaction.md [hunk 11]
```diff
@@ -38,31 +40,15 @@ Model.findAll({
...
-  include: [TaskModel, ...]
-}, {
-  transaction: t1,
-  lock: {
-    level: t1.LOCK...,
-    of: UserModel
-  }
-})
-```
-UserModel will be locked but TaskModel won't!
```
Comment (BridgeAR) [inline docs/api/transaction.md]: Why is this removed?

File: docs/api/model.md [hunk 12]
```diff
@@ -206,30 +221,32 @@ The success listener is called with an array of instances if the query succeeds.
...
 | [options.limit] | Number |  |
 | [options.offset] | Number |  |
 | [options.transaction] | Transaction |  |
-| [queryOptions] | Object | Set the query options, e.g. raw, specifying that you want raw data instead of built Instances. See sequelize.query for options |
-| [queryOptions.lock] | String|Object | Lock the selected rows. Possible options are transaction.LOCK.UPDATE and transaction.LOCK.SHARE. Postgres also supports transaction.LOCK.KEY_SHARE, transaction.LOCK.NO_KEY_UPDATE and specific model locks with joins. See [transaction.LOCK for an example](api/transaction#lock)  |
+| [options.lock] | String | Lock the selected rows in either share or update mode. Possible options are transaction.LOCK.UPDATE and transaction.LOCK.SHARE. See [transaction.LOCK for an example](https://github.com/sequelize/sequelize/wiki/API-Reference-Transaction#LOCK) |
```
Comment (BridgeAR) [inline docs/api/model.md]: The old description was actually the rigt one?

Comment (janmeier) [inline docs/api/model.md]: Hmm, the new description matches the one in the code though - it seems the code was not updated, only that docs?

Comment (BridgeAR) [inline docs/api/model.md]: True, I made a mistake changing the description in the wrong place. I'll fix that later / tomorrow.

File: docs/api/instance.md (hunk 10 above)
Comment (sdepold) [inline docs/api/instance.md]: arg

Comment (sdepold) [inline docs/api/instance.md]: fixed

File: docs/api/model.md (hunk 12 above)
Comment (sdepold) [inline docs/api/model.md]: rebased

**Comment by sdepold** [comment]
@janmeier @mickhansen @BridgeAR ptal

File: docs/api/datatypes.md [hunk 13]
```diff
@@ -89,7 +110,7 @@ Available properties: `UNSIGNED`, `ZEROFILL`
 
 <a name="decimal"></a>
 ## `DECIMAL()`
-[View code](https://github.com/sequelize/sequelize/blob/421f0f34356a3484b0f26e92e7fd133af6d3f6db/lib/data-types.js#L225)
+[View code](https://github.com/sequelize/sequelize/blob/716cb2d2aae3a4cd7fdaace13411cf4161e6deb6/lib/data-types.js#L263)
 Decimal number. Accepts one or two arguments for precision
 
 Available properties: `UNSIGNED`, `ZEROFILL`
```
Comment (BridgeAR) [inline docs/api/datatypes.md]: This is actually not true. But that's more a reminder to me and should not hold up the MR.

**Comment by BridgeAR** [comment]
LGTM

**Comment by janmeier** [comment]
Thanks, I was getting a bit behind with keeping the docs updated! :)

**Comment by sdepold** [comment]
:)
